Visits serializer giving 500 in overland endpoint

```

  File "/app/wayfinder/views.py", line 145, in post

    if visit_serializer.is_valid():

AssertionError: .validate() should return the validated data

```

--- a/backend/wayfinder/serializers.py
+++ b/backend/wayfinder/serializers.py
@@ -96,10 +96,6 @@ class VisitSerializer(serializers.ModelSerializer):
         geometry = data.get("geometry", {})
         coordinates = geometry.get("coordinates", [])
 
-        # If there is no departure date, ignore this visit
-        if not properties.get("departure_date"):
-            return None
-
         # Calculate the duration of the visit
         arrival_date = properties.get("arrival_date")
         departure_date = properties.get("departure_date")

--- a/backend/wayfinder/views.py
+++ b/backend/wayfinder/views.py
@@ -140,19 +140,18 @@ class OverlandView(APIView):
 
         for item in locations_data:
             if "arrival_date" in item.get("properties", {}):
+
+                # If there is no departure date, skip the visit
+                departure_date = item.get("properties", {}).get("departure_date")
+                if departure_date is None or departure_date == "":
+                    log.debug("Skipping visit without valid departure date")
+                    log.debug(f"Visit data: {item}")
+                    continue
+
                 # This is a visit
                 visit_serializer = VisitSerializer(data=item)
+
                 if visit_serializer.is_valid():
-
-                    # Check that the visit has a departure date
-                    # Visit data is first sent when the user arrives at a location, setting an arrival date but no departure date
-                    # When the user leaves the location, the same object with the departure date is sent
-                    # There is no need to save the first object, as it will be updated with the departure date when the second object is received
-                    if not visit_serializer.validated_data.get("departure_date"):
-                        log.debug("Skipping visit without departure date")
-                        log.debug(f"Visit data: {item}")
-                        continue
-
                     visits_to_create.append(Visit(**visit_serializer.validated_data))
                 else:
                     log.error(f"Visit validation error: {visit_serializer.errors}")

